### PR TITLE
feat: Multiselect warehouse filter added into batchwise balance history report

### PR DESCRIPTION
**Pull request by vishdha**
TASK ID: [ASANA](https://app.asana.com/0/1192407894480744/1199998672071088)
Peek 2021-03-02 16-22

### CODE REVIEW

File: erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py [hunk 1]
```diff
@@ -45,29 +45,30 @@ def get_columns(filters):
...
 
 def get_conditions(filters):
-	conditions = ""
+	conditions = []
 	if not filters.get("from_date"):
 		frappe.throw(_("'From Date' is required"))
 
 	if filters.get("to_date"):
-		conditions += " and posting_date <= '%s'" % filters["to_date"]
+		conditions.append("posting_date <= %(to_date)s")
```
Comment (sahil28297) [inline erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py]:
```suggestion
		conditions.append(" and posting_date <= %(to_date)s")
```

File: erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py [hunk 2]
```diff
@@ -45,29 +45,30 @@ def get_columns(filters):
...
 	else:
 		frappe.throw(_("'To Date' is required"))
 
-	return conditions
+	if filters.get("warehouse"):
+		filters.warehouse = frappe.parse_json(filters.get('warehouse'))
+		conditions.append("warehouse in %(warehouse)s")
```
Comment (sahil28297) [inline erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py]:
```suggestion
		conditions.append(" and warehouse in %(warehouse)s")
```

File: erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py [hunk 3]
```diff
@@ -45,29 +45,30 @@ def get_columns(filters):
...
 
+	return "and {}".format(" and ".join(conditions)) if conditions else ""
```
Comment (sahil28297) [inline erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py]:
```suggestion
	return "".join(conditions) if conditions else ""
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]

File: erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py (hunk 2 above)
Comment (vishdha) [reply erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py]: @sahil28297 this is not a string. this is the list now just mention "and" gives error incorrect syntax thats why we need to join

File: erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py [hunk 4]
```diff
@@ -45,29 +45,30 @@ def get_columns(filters):
...
 
 def get_conditions(filters):
-	conditions = ""
+	conditions = []
 	if not filters.get("from_date"):
 		frappe.throw(_("'From Date' is required"))
 
 	if filters.get("to_date"):
-		conditions += " and posting_date <= '%s'" % filters["to_date"]
+		conditions.append("and posting_date <= %(to_date)s")
```
Comment (sahil28297) [inline erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py:53]:
```suggestion
		conditions.append(" and posting_date <= %(to_date)s")
```

File: erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py [hunk 5]
```diff
@@ -45,29 +45,30 @@ def get_columns(filters):
...
 	else:
 		frappe.throw(_("'To Date' is required"))
 
-	return conditions
+	if filters.get("warehouse"):
+		filters.warehouse = frappe.parse_json(filters.get('warehouse'))
+		conditions.append("and warehouse in %(warehouse)s")
```
Comment (sahil28297) [inline erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py]:
```suggestion
		conditions.append(" and warehouse in %(warehouse)s")
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]

File: erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py (hunk 4 above)
Comment (vishdha) [reply erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py:53]: @sahil28297 this space not needed. oterwise two space one from here and anoter from join will get added

Comment (sahil28297) [reply erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py:53]: join doesn't add a space, but there's already a space added in the sql statement before condition so yeah, we don't need it. Also 2 spaces in sql statement doesn't throw any error.

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**